### PR TITLE
[agent] use OpenThread daemon

### PR DESCRIPTION
**Pull request by bukepo**
* add a new implementation of `Ncp::Controller` based on OpenThread daemon mode
* add support in web GUI
* add CI tests for this new mode

### CODE REVIEW

File: src/agent/ncp.hpp [hunk 1]
```diff
@@ -134,15 +136,15 @@ class Controller : public EventEmitter
      * @param[in]   aInterfaceName  A string of the NCP interface.
```
Comment (gjc13) [inline src/agent/ncp.hpp]: Docs for new arguments

File: src/agent/ncp_openthread.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "ncp_posix.hpp"
```
Comment (gjc13) [inline src/agent/ncp_openthread.cpp]: Should it be `ncp_openthread.hpp`?

File: src/agent/ncp_openthread.hpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+     * @retval  OTBR_ERROR_NONE         Successfully requested the event.
+     * @retval  OTBR_ERROR_ERRNO        Failed to request the event.
+     *
+     */
+    virtual otbrError RequestEvent(int aEvent);
+
+    virtual ~ControllerPosix(void);
+
+private:
+    static void HandleStateChanged(otChangedFlags aFlags, void *aContext)
```
Comment (gjc13) [inline src/agent/ncp_openthread.hpp]: Where is this `HandleStateChanged` called?

File: src/agent/ncp_openthread.hpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+namespace Ncp {
+
+/**
+ * This interface defines NCP Controller functionality.
+ *
+ */
+class ControllerOpenThread : public Controller
+{
+public:
```
Comment (gjc13) [inline src/agent/ncp_openthread.hpp]: Doc string

File: third_party/openthread/mbedtls-config.h [hunk 5]
```diff
@@ -79,10 +80,17 @@
 /* Save RAM by adjusting to our exact needs */
 #define MBEDTLS_ECP_MAX_BITS 256
 #define MBEDTLS_MPI_MAX_SIZE 32 // 256 bits is 32 bytes
+#define MBEDTLS_SSL_MAX_CONTENT_LEN 900
 
 /* Save ROM and a few bytes of RAM by specifying our own ciphersuite list */
 #define MBEDTLS_SSL_CIPHERSUITES MBEDTLS_TLS_ECJPAKE_WITH_AES_128_CCM_8
 
+// #define MBEDTLS_MEMORY_BUFFER_ALLOC_C
```
Comment (gjc13) [inline third_party/openthread/mbedtls-config.h]: remove unused defines?

File: src/web/web-service/ot_client.cpp [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+    count = write(mSocket, mBuffer, ret);
+
+    if (count < ret)
+    {
+        mBuffer[ret] = '\0';
+        otbrLog(OTBR_LOG_ERR, "Failed to send command: %s", mBuffer);
+    }
+
+    for (int i = 0; i < mTimeout; ++i)
```
Comment (gjc13) [inline src/web/web-service/ot_client.cpp]: Why not select with the total timeout?

Comment (bukepo) [reply src/web/web-service/ot_client.cpp]: `select()` will return the moment the fd is readable, so read() may not receive all outputs. Especially for `scan`

File: .gitmodules [hunk 7]
```diff
@@ -1,3 +1,3 @@
 [submodule "third_party/openthread/repo"]
 	path = third_party/openthread/repo
-	url = https://github.com/openthread/openthread.git
+	url = https://github.com/bukepo/openthread.git
```
Comment (bukepo) [inline .gitmodules]: Will update this after openthread/openthread#3591 is merged.

File: src/web/web-service/ot_client.cpp [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+        ++rval;
+    }
+
+    mTimeout = kDefaultTimeout;
+
+exit:
+    return rval;
+}
+
+bool Client::FactoryReset()
```
Comment (gjc13) [inline src/web/web-service/ot_client.cpp]: FactoryReset(void)

File: src/web/web-service/ot_client.cpp [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+{
+    const char *result;
+    bool        rval = false;
+
+    Execute("factoryreset");
+    sleep(0);
```
Comment (gjc13) [inline src/web/web-service/ot_client.cpp]: `yield()` instead?

File: src/web/web-service/ot_client.cpp [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+#include <sys/un.h>
+#include <unistd.h>
+
+#include "platform-posix.h"
+#include "common/code_utils.hpp"
+#include "common/logging.hpp"
+
+namespace ot {
+
+Client::Client(void)
```
Comment (gjc13) [inline src/web/web-service/ot_client.cpp]: Shall we make `OPENTHREAD_POSIX_APP_SOCKET_NAME` and `timeout` configurable in constructor and make the current values default parameter?

**Pull request merged**